addons/survey/models/survey_question.py

```diff
@@ -312,6 +312,18 @@ def _compute_is_scored_question(self):
             else:
                 question.is_scored_question = False
 
+    @api.onchange('question_type', 'validation_required')
+    def _onchange_validation_parameters(self):
+        """Ensure no value stays set but not visible on form, preventing saving."""
+        self.validation_min_float_value = 0
+        self.validation_max_float_value = 0
+        self.validation_length_min = 0
+        self.validation_length_max = 0
+        self.validation_min_date = False
+        self.validation_max_date = False
+        self.validation_min_datetime = False
+        self.validation_max_datetime = False
```

Comment:
so if you change question_type only, for example, text to char box, you reset the length anyway ?
sad :cry:

Reply:
It shouldn't be a problem since all of these fields are only used for a single question type, including length which only applies to char box.